### ISSUE TITLE
Including a p5.Vector in saveJSON throws a circular structure error

### ISSUE THREAD

**Issue by chrsmlls333** [processing/p5.js#5470]
Submitting an object with a p5.Vector object throws a TypeError. Test code and associated error:
```js
let v = createVector(4,4);
saveJSON({ v }, 'test');
```
```
Uncaught TypeError: Converting circular structure to JSON
    --> starting at object with constructor 'p5'
    |     property '_curElement' -> object with constructor '_main.default.Renderer2D'
    --- property '_pixelsState' closes the circle
    at JSON.stringify (<anonymous>)
    at p5._main.default.saveJSON (p5.js:85038)
    at <anonymous>:1:1
```
  
   
## I think two solutions are possible:

1. ### Filter out any p5 object properties transparent to the end user.
    This would involve assessing p5.Vector and any other built-in classes for error-throwing content and making a custom replacer function to replace these with simplified or string equivalents.

2. ### Expand the saveJSON arguments to include a user-provided replacer.
    Here is some possible code for this:

```
p5.prototype.saveJSON = function(json, filename, opt, replacer) {
  p5._validateParameters('saveJSON', arguments);
  let stringify;
  if (opt) {
    stringify = JSON.stringify(json, replacer);
  } else {
    stringify = JSON.stringify(json, replacer, 2);
  }
  this.saveStrings(stringify.split('\n'), filename, 'json');
};
```

3. ### Both of the above
   We could make a basic replacer and then allow a user to provide their own to either compound or override the filter.

---
Location of code edit:
https://github.com/processing/p5.js/blob/cb3f59740bcb71ba5e2bee820bb1f784004d85ef/src/io/files.js#L1578

**Comment by chrsmlls333** [comment]
Oh, there is already a pull request to solve the vector issue specifically. https://github.com/processing/p5.js/pull/5469

Still, this may be present for other structures?

**Comment by limzykenneth** [comment]
#5469 will solve this problem for p5.Vector. Cyclic reference is still in the p5 instance itself and it will not be possible to be refactor out easily. I haven't come across anywhere else where it is a problem yet but if you find one let us know and we can look into it.

**Comment by chrsmlls333** [comment]
Would it make sense to filter out p5 instances here regardless? That could be a blanket fix for beginners' accessibility.

**Comment by limzykenneth** [comment]
What do you mean?

**Comment by limzykenneth** [comment]
Resolved by #5469
